data/alsa_drivers.rb

```diff
@@ -107,15 +106,27 @@ def params
 
   # find all sound drivers below the given path
   def self.find_all(path)
-    ret = []
-    lst = `find #{path} -type f -name 'snd-*.ko'`.split("\n").sort{|p1, p2| 
-	p1.split('/').last <=> p2.split('/').last
-    }
+    result = []
 
-    lst.each do |m|
-      ret << AlsaModule.new(m)
-    end
+    output = Command.output("find", path, "-type", "f", "-name", "snd-*.ko")
 
-    ret
+    drivers = output.split("\n").sort { |p1, p2| p1.split('/').last <=> p2.split('/').last }
+
+    drivers.map { |d| AlsaModule.new(d) }
+  end
+
+  # Auxiliary class to run a command and get its output
+  class Command
+    # Returns the output of the given command
+    #
+    # @param args [Array<String>, Array<Array<String>>] the command to execute and
+    #   its arguments. For a detailed description, see
+    #   https://www.rubydoc.info/github/openSUSE/cheetah/Cheetah#run-class_method
+    # @return [String] commmand output or an empty string if the command fails.
+    def self.output(*args)
```

Comment:
Now that we have the arguments so nicely separated already, maybe split off the command (`shift`) and shell-escape all the others?

Reply:
That is automatically done by `Yast::Execute`, it escapes the arguments, so not needed here.

Reply:
OK then.